src/inmanta/module.py

```diff
@@ -1155,6 +1155,45 @@ def _remove_line_continuations(self, lines: List[str]) -> List[str]:
                     result.append(line)
         return result
 
+    def _get_requirements_txt_as_list(self) -> List[str]:
+        """
+        Returns the contents of the requirements.txt file as a list of requirements, if it exists.
+        """
+        file = os.path.join(self._path, "requirements.txt")
+        if os.path.exists(file):
+            with open(file, "r", encoding="utf-8") as fd:
+                requirements_txt_content = fd.read()
+                req_lines = [x for x in requirements_txt_content.split("\n") if len(x.strip()) > 0]
+                req_lines = self._remove_comments(req_lines)
+                req_lines = self._remove_line_continuations(req_lines)
+                return list(set(req_lines))
+        else:
+            return []
+
+    @abstractmethod
+    def get_all_python_requirements_as_list(self) -> List[str]:
+        """
+        Returns all Python requirements specified by this module like, including requirements on V2 modules.
+        """
+        raise NotImplementedError()
+
+    def get_strict_python_requirements_as_list(self) -> List[str]:
+        """
+        Returns the strict python requirements specified by this module like, meaning all Python requirements excluding those on
+        inmanta modules.
+        """
+        return [req for req in self.get_all_python_requirements_as_list() if not req.startswith(ModuleV2.PKG_NAME_PREFIX)]
+
+    def get_module_v2_requirements(self) -> List[InmantaModuleRequirement]:
+        """
+        Returns all requirements this module like has on v2 modules.
+        """
+        return [
+            InmantaModuleRequirement.parse(ModuleV2Source.get_inmanta_module_name(req))
+            for req in self.get_all_python_requirements_as_list()
+            if req.startswith(ModuleV2.PKG_NAME_PREFIX)
+        ]
+
```

Comment:
Nothing really new here. I moved some methods up to `ModuleLike` so `Project` has them as well and I added `get_module_v2_requirements`.